File: AndroidStudioProjects/EjercicioFragments_Toolbar/app/src/main/java/com/example/ejerciciofragments_toolbar/MainActivity.java
package com.example.ejerciciofragments_toolbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.os.Bundle;
import android.view.Menu;

public class MainActivity extends AppCompatActivity {

    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Añadimos toolbar
        toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        //Inflamos el menu con el toolbar
        getMenuInflater().inflate(R.menu.toolbar, menu);
        return true;
    }
}

File: AndroidStudioProjects/Fragments/app/src/main/java/com/example/fragments/MainActivity.kt
package com.example.fragments

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), OnFragmentActionListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnPink.setOnClickListener() {
            loadFragment(PinkFragment())
        }

        btnBlue.setOnClickListener() {
            loadFragment(BlueFragment())
        }
    }

    //Implementamos OnFragmentActionListener en la clase y se crea este método
    override fun OnClickFragmentButton() {
        Toast.makeText(this, "El botón ha sido pulsado", Toast.LENGTH_SHORT).show()
    }

    //Transmitimos los cambios del Fragment al layout creado para el fragment
    private fun loadFragment(fragment: Fragment) {
        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.add(R.id.fragmentContainer, fragment)
        fragmentTransaction.commit()
    }

}